src/components/WindowChildren/WindowTerminal.vue

```diff
@@ -104,7 +124,10 @@ export default {
       let words = this.input_text.substring(this.protection_length,this.input_text.length);
       this.protection_length += words.length
       words = words.replace(/^\s\s*/, '').replace(/\s\s*$/, '')
-      
+      //add to command history
+      if(words!=''&&(words.length==0||words.length>0&&words!=this.history_cmd_stack[-1])){
+      this.history_cmd_stack.length>=this.HISTSIZE&&this.history_cmd_stack.pop()
+      this.history_cmd_stack.unshift(words)}
```

Comment:
Don't understand the point of `words!=''&& words.length==0` could you explain, and I think it's better to add parenthesis like
```
if ( words!='' && ( words.length==0 || (words.length>0 && words!=this.history_cmd_stack[-1]))) {
  if (this.history_cmd_stack.length>=this.HISTSIZE) {
    this.history_cmd_stack.pop()
  }
  this.history_cmd_stack.unshift(words)
}
```

Reply:
And I think `HISTSIZE` can be a larger value, like 99 or more, as we are not concerned about the system resources it takes up.

Reply:
😂Sorry,  it just because I confused myself at that time